code/game/objects/structures.dm

```diff
@@ -9,6 +9,7 @@
 
 	var/damage = 0
 	var/damage_cap = 500 //The point where things start breaking down.
+	var/health
```

Comment:
While it's positive to remove the snowflakiness of the barricade health system, then we'd have both `health` and `damage` for structures.
The best answer here is to emulate /tg/ and define `obj_integrity` and `max_integrity` at the `/obj` level, and slowly transition to it.

Reply:
Yes at some point that'll get done, or a component will get written